File: 001_reading_displaying_write_image/Python/read_display_write_opencv.py
# The python libraries numpy and cv2 need to be imported before reading an image.
import numpy as np
import cv2

# The function cv2.imread() is used to read an image.
img_color = cv2.imread('test.jpg',1)
img_grayscale = cv2.imread('test.jpg',0)
img_unchanged = cv2.imread('test.jpg',-1)


# The function cv2.imshow() is used to display an image in a window.
cv2.imshow('color image',img_color)
cv2.imshow('grayscale image',img_grayscale)
cv2.imshow('unchanged image',img_unchanged)

#Printing the image type and shape for Color Image
print(type(img_color))
print(img_color.shape)

# waitKey() waits for a key press to close the window and 0 specifies indefinite loop
cv2.waitKey(0)

# cv2.destroyAllWindows() simply destroys all the windows we created.
cv2.destroyAllWindows()

# The function cv2.imwrite() is used to write an image.
cv2.imwrite('grayscale.jpg',img_grayscale)

File: 001_reading_displaying_write_image/CPP/read_display_write_opencv.cpp
// Include Libraries
#include<opencv2/opencv.hpp>
#include<iostream>

// Namespace nullifies the use of cv::function();
using namespace std;
using namespace cv;

int main()
{
	// Read image
	Mat img_color, img_grayscale, img_unchanged;
	img_color = imread("test.jpg",1);
	img_grayscale = imread("test.jpg",0);
	img_unchanged = imread("test.jpg",-1);

	// Print image type and shape
	cout << "Color Image type : " << img_color.type() << endl;
	cout << "Color Image shape : " << img_color.size() << endl;


	//display image
	imshow("color image",img_color);
	imshow("grayscale image",img_grayscale);
	imshow("unchanged image",img_unchanged);

	//Write the image in the same directory
	//We are saving the same image here without making any changes
	//Just to understand how image saving is carried out

	imwrite("grayscale_image.jpg",img_grayscale);

	// 0 means loop infinitely
	// Press any key to close it

	waitKey(0);
	return 0;
}
